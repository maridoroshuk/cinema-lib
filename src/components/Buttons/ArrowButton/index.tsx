import React from 'react';

import ArrowImage from '@/assets/icons/arrow.png';

import { Arrow, Wrapper } from './styles';
import { IArrowButton } from './interface';

export const ArrowButton = ({ onClick }: IArrowButton) => {
  return (
    <Wrapper onClick={onClick}>
      <Arrow src={ArrowImage} alt="arrow (hide)" />
    </Wrapper>
  );
};
