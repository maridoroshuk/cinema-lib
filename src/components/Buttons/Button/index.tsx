import React from 'react';

import { ButtonStyled } from '@/components/Buttons/Button/styles';

import { IButtonProps } from './interface';

export const Button = ({
  children,
  size,
  backgroundColor,
  type,
  value,
  onClick,
  dataCy,
}: IButtonProps) => {
  return (
    <ButtonStyled
      backgroundColor={backgroundColor}
      size={size}
      type={type || 'button'}
      value={value}
      onClick={onClick}
      data-cy={dataCy}
    >
      {children}
    </ButtonStyled>
  );
};
