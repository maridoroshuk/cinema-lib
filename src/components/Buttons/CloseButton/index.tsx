import React from 'react';
import { ICloseButtonProps } from './interface';
import { ButtonWrapper, StyledButton } from './styles';

export const CloseButton = ({ onClose }: ICloseButtonProps) => {
  return (
    <ButtonWrapper>
      <StyledButton onClick={onClose} />
    </ButtonWrapper>
  );
};
