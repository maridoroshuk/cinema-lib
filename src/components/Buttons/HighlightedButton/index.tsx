import React from 'react';
import { IUnderlineButton } from './interface';
import { ButtonStyled } from './styles';

export const HighlightedButton = ({
  children,
  handleButtonClick,
  value,
}: IUnderlineButton) => {
  return (
    <ButtonStyled onClick={handleButtonClick} value={value || ''}>
      {children}
    </ButtonStyled>
  );
};
