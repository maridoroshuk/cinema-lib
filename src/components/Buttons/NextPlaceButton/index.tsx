import React from 'react';

import NextPlaceButton from '@/assets/icons/next_place_button.png';

import { Next, Wrapper } from './styles';

export interface IArrowButton {
  onClick?: () => void;
}

export const NextPlace = ({ onClick }: IArrowButton) => {
  return (
    <Wrapper onClick={onClick}>
      <Next src={NextPlaceButton} />
    </Wrapper>
  );
};
