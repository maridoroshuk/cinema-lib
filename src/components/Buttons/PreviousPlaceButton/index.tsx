import React from 'react';

import PreviousPlaceButton from '@/assets/icons/previous_place_button.png';

import { Previous, Wrapper } from './styles';

export interface IArrowButton {
  onClick?: () => void;
}

export const PreviousPlace = ({ onClick }: IArrowButton) => {
  return (
    <Wrapper onClick={onClick}>
      <Previous src={PreviousPlaceButton} />
    </Wrapper>
  );
};
