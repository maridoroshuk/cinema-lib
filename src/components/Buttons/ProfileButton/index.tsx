import React from 'react';
import { IProfileButton } from './interface';
import { ButtonStyled } from './styles';

export const ProfileButton = ({ onClick }: IProfileButton) => {
  return <ButtonStyled onClick={onClick}>Profile</ButtonStyled>;
};
