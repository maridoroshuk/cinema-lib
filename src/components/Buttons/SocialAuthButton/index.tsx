import React from 'react';

import { facebookStyles, githubStyles, googleStyles } from './config';
import { ISocialAuthButtonProps } from './interface';
import { LogoButton } from './LogoButton';

export const SocialAuthButton = ({
  variant,
  handleButtonClick,
  children,
}: ISocialAuthButtonProps) => {
  return (
    <>
      {variant === 'google' ? (
        <LogoButton
          config={googleStyles}
          value={variant}
          onClick={handleButtonClick}
        >
          {children}
        </LogoButton>
      ) : null}
      {variant === 'facebook' ? (
        <LogoButton
          config={facebookStyles}
          value={variant}
          onClick={handleButtonClick}
        >
          {children}
        </LogoButton>
      ) : null}
      {variant === 'github' ? (
        <LogoButton
          config={githubStyles}
          value={variant}
          onClick={handleButtonClick}
        >
          {children}
        </LogoButton>
      ) : null}
    </>
  );
};
