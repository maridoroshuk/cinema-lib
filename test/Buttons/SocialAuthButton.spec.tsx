import React from 'react';
import { render } from '@testing-library/react';

import { SocialAuthButton } from '@/components/Buttons/SocialAuthButton';

import 'jest-styled-components';

import { setupTestProvider } from '../setupTests';

describe('SocialAuthButton', () => {
  it('renders google button', () => {
    const handleClick = jest.fn();
    const { container } = render(
      setupTestProvider(
        <SocialAuthButton variant="google" handleButtonClick={handleClick} />,
      ),
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders facebook button', () => {
    const handleClick = jest.fn();
    const { container } = render(
      setupTestProvider(
        <SocialAuthButton variant="facebook" handleButtonClick={handleClick} />,
      ),
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders github button', () => {
    const handleClick = jest.fn();
    const { container } = render(
      setupTestProvider(
        <SocialAuthButton variant="github" handleButtonClick={handleClick} />,
      ),
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
