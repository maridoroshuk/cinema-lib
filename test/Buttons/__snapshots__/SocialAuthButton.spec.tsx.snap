// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialAuthButton renders facebook button 1`] = `
.c0 {
  width: 100%;
  padding: 15px 30px;
  color: #FFFFFF;
  background-color: #1877F2;
  font-family: 'Poppins',sans-serif;
  font-weight: 600;
  font-size: 14px;
  border: none;
  border-radius: 10px;
  cursor: pointer;
  text-align: center;
  margin: 0 auto;
}

.c0:before {
  width: 20px;
  height: 20px;
  content: '';
  display: inline-block;
  object-fit: contain;
  vertical-align: bottom;
  background-image: url();
  background-size: cover;
  margin-right: 20px;
}

@media (min-width:860px) {
  .c0 {
    max-width: 340px;
    width: 100%;
  }
}

<button
  class="c0"
  value="facebook"
>
  Sign up with Facebook
</button>
`;

exports[`SocialAuthButton renders github button 1`] = `
.c0 {
  width: 100%;
  padding: 15px 30px;
  color: #FFFFFF;
  background-color: #000000;
  font-family: 'Poppins',sans-serif;
  font-weight: 600;
  font-size: 14px;
  border: none;
  border-radius: 10px;
  cursor: pointer;
  text-align: center;
  margin: 0 auto;
}

.c0:before {
  width: 20px;
  height: 20px;
  content: '';
  display: inline-block;
  object-fit: contain;
  vertical-align: bottom;
  background-image: url();
  background-size: cover;
  margin-right: 20px;
}

@media (min-width:860px) {
  .c0 {
    max-width: 340px;
    width: 100%;
  }
}

<button
  class="c0"
  value="github"
>
  Sign up with GitHub
</button>
`;

exports[`SocialAuthButton renders google button 1`] = `
.c0 {
  width: 100%;
  padding: 15px 30px;
  color: #0000008c;
  background-color: #FFFFFF;
  font-family: 'Poppins',sans-serif;
  font-weight: 600;
  font-size: 14px;
  border: none;
  border-radius: 10px;
  cursor: pointer;
  text-align: center;
  margin: 0 auto;
}

.c0:before {
  width: 20px;
  height: 20px;
  content: '';
  display: inline-block;
  object-fit: contain;
  vertical-align: bottom;
  background-image: url();
  background-size: cover;
  margin-right: 20px;
}

@media (min-width:860px) {
  .c0 {
    max-width: 340px;
    width: 100%;
  }
}

<button
  class="c0"
  value="google"
>
  Continue with Google
</button>
`;
